transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -89,7 +91,8 @@ public SimpleChannelPool(Bootstrap bootstrap, final ChannelPoolHandler handler,
      */
     public SimpleChannelPool(Bootstrap bootstrap, final ChannelPoolHandler handler, ChannelHealthChecker healthCheck,
                              boolean releaseHealthCheck) {
-        this.handler = checkNotNull(handler, "handler");
+        ChannelPoolStats statsHandler = new ChannelPoolStats(checkNotNull(handler, "handler"));
```

Comment:
if we are going ahead with this PR, can we lazy or conditionally create the `ChannelPoolStats` wrapper?